### PR TITLE
fix flashinit panic not printing

### PR DESCRIPTION
**Pull request by mhightower83**
Resolves
* https://github.com/esp8266/Arduino/issues/8761#issue-1499638772

### CODE REVIEW

File: cores/esp8266/flash_hal.h [hunk 1]
```diff
@@ -33,21 +33,32 @@ extern "C" {
...
 
+#ifndef QUOTE
+#define QUOTE(a) __STRINGIFY(a)
+#endif
+
+#ifdef DEBUG_ESP_CORE
+#define DEBUG_FLASH_MAP_NOT_FOUND " flashinit: configuration not found"
+#else
+#define DEBUG_FLASH_MAP_NOT_FOUND
+#endif
```
Comment (mcspr) [inline cores/esp8266/flash_hal.h:48]: we always print exc text though, why hide?
could also just have `static const char DEBUG_FLASH_MAP_NOT_FOUND[] PROGMEM = ...` right here at the top?

Comment (mhightower83) [reply cores/esp8266/flash_hal.h:48]: Was concerned about growing space used up in non-debug builds.

I was thinking I wanted the line # reported to be in the function body; however, I later realized it is the macro line number that is being reported. Also, if I place it in the scope of the function I avoid the potential problem of a hidden variable name conflict in the module.

I have no strong preferences.

File: cores/esp8266/core_esp8266_main.cpp [hunk 2]
```diff
@@ -480,6 +480,11 @@ extern "C" void user_init(void) {
 
     uart_div_modify(0, UART_CLK_FREQ / (115200));
 
+#if FLASH_MAP_SUPPORT
+    const char *err_msg = flashinit();
+    if (err_msg) __unhandled_exception(err_msg);
```
Comment (mhightower83) [inline cores/esp8266/core_esp8266_main.cpp:485]: The message text looks like this when printed. Is there any objection to the "Unhandled C++ exception" choice in the postmortem? I didn't see any others that would work well for a single string.
```
Unhandled C++ exception: autoconfig.ino.ino:15 flashinit: configuration not found
```

Comment (d-a-v) [reply cores/esp8266/core_esp8266_main.cpp:485]: could `__panic__func()` or `__assert_func()` help ?

Comment (mhightower83) [reply cores/esp8266/core_esp8266_main.cpp:485]: If I change postmortem to print panics when `s_panic_line` is set to 0. That seems to work. I'll push the change.

File: cores/esp8266/flash_hal.h (hunk 1 above)
Comment (d-a-v) [reply cores/esp8266/flash_hal.h:48]: Maybe file and line are unnecessary in this case, and only the string is relevant and useful even in non-debug mode.

Comment (mhightower83) [reply cores/esp8266/flash_hal.h:48]: That was my first attempt then I realized the user could provide a broken override `flashinit`. 
Providing the file and line gives a strong indication of where the fault lives. And that the override won during the linking process.

**Comment by mcspr** [comment]
Just as a thought, don't we want to return something like this?

```cpp
FlashInitResult flashinit();
```
```cpp
struct FlashInitResult {
    bool ok { false };
    const char* err_msg;
};
```

**Comment by d-a-v** [comment]
`flashinit()` was not designed to fail, so it is 'panic'ing instead

**Comment by mhightower83** [comment]
Too many merge conflicts were corrected with web edit, I need to correct some stuff and do a few real test.

**Comment by mcspr** [comment]
> flashinit() was not designed to fail, so it is 'panic'ing instead

I just mean the return value. Struct member name tells us what we mean right there, no need to remember triple meaning of a char pointer

**Review by d-a-v — APPROVED** [review]

**Comment by mhightower83** [comment]
> `FlashInitResult flashinit();`

FWIF: I tried this method and it increases "IROM code in flash" by 16 bytes.

**Pull request merged**